reliquery/storage.py

```diff
@@ -505,6 +511,463 @@ def get_all_relic_data(self) -> List[Dict]:
             return relic_data
 
 
+class GoogleDriveStorage(Storage):
+    def __init__(
+        self,
+        prefix: str,
+        name: str,
+        token_file: str,
+        SCOPES: list,
+        shared_folder_id: str,
+    ):
+        self.prefix = prefix
+        self.name = name
+        self.token_file = token_file
+        self.SCOPES = SCOPES
+        self.shared_folder_id = shared_folder_id
+
+        creds = None
+        if os.path.exists(token_file):
+            creds = service_account.Credentials.from_service_account_file(
+                token_file, scopes=SCOPES
+            )
+        try:
+            self.service = build("drive", "v3", credentials=creds)
+        except HttpError as error:
```

Comment:
We might want to consider letting this exception bubble up. If something goes wrong with the build method we will only want to catch it if there is something we can actually do about the error. My understanding is that if the build fails we can't use any of the logic google api has to offer anyways. I would say that the only time we should catch error is in in the case were we have defensive logic to handle it or we want to raise another exception.
```python
except IOError:
            raise StorageItemDoesNotExist
```

Let  me know if this makes sense.